Add support for Rust-based tools

jsf9k opened the pull request:
## 🗣 Description ##

This pull request adds support for Rust-based tools to this Ansible role.

## 💭 Motivation and context ##

Resolves #41.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
Use a monospace font for pip

jsf9k committed:
Add support for Rust-based tools

jsf9k committed:
Add a Molecule scenario to test the Rust support

jsf9k committed:
Rename install_cargo.yml to install_rust.yml

jsf9k committed:
Alphabetize Ansible task attributes

jsf9k committed:
Remove Debian Buster and Bullseye from the Rust scenario

Neither version of Debian has a new enough version of cargo to support
the aardwolf dependency of CrackMapExec, even in backports.

jsf9k committed:
Add the ability to build a project with cargo

Also remove the explicit installation of rustc, since it will be
pulled in as a dependency of cargo.

jsf9k committed:
Change the project installed in the converge playbook of the rust scenario

CrackMapExec is the reason this functionality is being added, since it
is a Python tool with no wheel available that requires a Rust compiler
to build its native code.  Unfortunately, CrackMapExec does not make a
good test case.  The reason is that some of its dependencies are not
versioned but instead just pull in the HEAD of the default branch.
Such an installation can never pass idempotence.

Since the new functionality allows for the building of a Rust-based
project is makes sense to go ahead and test that use case.

dav3r reviewed (approved):
Nice improvement! ⛵

mcdonnnj commented on molecule/rust/molecule.yml:
```diff
@@ -0,0 +1,88 @@
+  # Debian Bullseye does not have a new enough version of cargo to
+  # support the aardwolf dependency of CrackMapExec, even in
+  # backports.
+  # - name: debian10
+  #   image: debian:buster-slim
```
This appears to accidentally be duplicated.

```suggestion
```

jsf9k replied:
Please see commit 0ad4031.

mcdonnnj commented on molecule/rust/molecule.yml:
```diff
@@ -0,0 +1,88 @@
+  ansible-lint
+  flake8
+platforms:
+  # Debian Buster does not have a new enough version of cargo to
+  # support the aardwolf dependency of CrackMapExec, even in
+  # backports.
```
Since the converge configuration uses [RustScan](https://github.com/RustScan/RustScan) do these comments for disabled platforms still stand?

jsf9k replied:
Good point.  Please see commit 7dd16e8.

mcdonnnj reviewed (approved):
Excellent addition! We can now support that 🔥💩! I have a suggestion and a question for your consideration.
![image](https://user-images.githubusercontent.com/50747025/201988824-287fd875-67db-43e1-9e7c-e118e01d8e2d.png)

jsf9k committed:
Remove duplicate lines

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Re-add Ubuntu 18 to the rust scenario

Now that converge.yml installs RustScan instead of CrackMapExec,
Ubuntu 18 functions again.

Also update the comments that describe why Debian Buster and Bullseye
are commented out, since with the CrackMapExec to RustScan change they
still fail but for different reasons.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>
